modules/resolve-styles.js

```diff
@@ -137,6 +137,14 @@ var resolveStyles = function (
   // will not be here, so each component needs to use Radium.
   var newChildren = null;
   var oldChildren = renderedElement.props.children;
+  var resolveChild = function (child) {
+    // test if child is a valid ReactDOMElement and not ReactComponentElement
+    if (React.isValidElement(child) && typeof child.type === 'string') {
```

Comment:
Instead of the extra closure, which I find a bit hard to follow, let's just invert this check and return early from resolveStyles directly after the children resolution:

```js
if (!React.isValidElement(renderedElement) || typeof renderedElement.type !== 'string') {
  return renderedElement;
}
```

The check must come after child resolution because you could pass dom elements as children of custom components, and you'd want them to get resolved correctly. Please add a test like the following: 

```as
var renderedElement = (
  <div>
    <CustomComponent style={style}>
      <div style={style} />
    </CustomComponent>
  </div>
);
```

Also, once you add the check, you can remove it from inside the callback in React.Children.map.